views/007_events.sql

```diff
@@ -161,31 +161,61 @@ AFTER UPDATE ON checks
 FOR EACH ROW
 EXECUTE PROCEDURE insert_check_updates_in_event_queue ();
 
--- Insert component status updates in event_queue
-CREATE OR REPLACE FUNCTION insert_component_status_updates_in_event_queue() RETURNS TRIGGER AS $$
-DECLARE event_name TEXT;
+-- Insert config health updates in event_queue
+CREATE OR REPLACE FUNCTION insert_config_health_updates_in_event_queue ()
+RETURNS TRIGGER AS $$
+DECLARE
+    event_name TEXT;
 BEGIN
-    IF OLD.status = NEW.status THEN
+    IF OLD.health = NEW.health OR (OLD.health IS NULL AND NEW.health IS NULL) THEN
       RETURN NULL;
     END IF;
 
-    IF NEW.status = 'unknown' THEN
-      RETURN NULL;
-    END If;
+    event_name := CONCAT('config.health.', COALESCE(NULLIF(NEW.health, ''), 'unknown'));
 
-    event_name := 'component.status.' || NEW.status;
+    INSERT INTO event_queue(name, properties)
+    VALUES (event_name, jsonb_build_object('id', NEW.id))
+    ON CONFLICT (name, properties) DO UPDATE SET
+        created_at = NOW(),
+        last_attempt = NULL,
+        attempts = 0;
 
+    RETURN NULL;
+END
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER config_health_event_enqueue
+AFTER UPDATE ON config_items
+FOR EACH ROW
+EXECUTE PROCEDURE insert_config_health_updates_in_event_queue();
+
+-- Cleanup
+DROP TRIGGER IF EXISTS  component_status_enqueue ON components;
+DROP FUNCTION IF EXISTS insert_component_status_updates_in_event_queue;
+
+-- Insert component health updates in event_queue
+CREATE OR REPLACE FUNCTION insert_component_health_updates_in_event_queue()
+RETURNS TRIGGER AS $$
+DECLARE
+    event_name TEXT;
+BEGIN
+    IF OLD.health = NEW.health OR (OLD.health IS NULL AND NEW.health IS NULL) THEN
+      RETURN NULL;
+    END IF;
+
+    event_name := CONCAT('component.', COALESCE(NULLIF(NEW.health, ''), 'unknown'));
+    
     INSERT INTO event_queue (name, properties) VALUES (event_name, jsonb_build_object('id', NEW.id))
     ON CONFLICT (name, properties) DO UPDATE SET created_at = NOW(), last_attempt = NULL, attempts = 0;
 
     RETURN NULL;
 END
 $$ LANGUAGE plpgsql;
 
-CREATE OR REPLACE TRIGGER component_status_enqueue
```

Comment:
I think we would need a drop trigger if exists as well

Reply:
It's there

https://github.com/flanksource/duty/pull/734/files#diff-5f7fb935294a31e0296a95edbb172d0f7e9b4e84f7578e488d91d52760c0f0c2R193-R194